recipes/repo.rb

```diff
@@ -25,7 +25,9 @@
 
 # Set defaults for version, based on distribution
 node.default['hadoop']['distribution_version'] =
-  if node['hadoop']['distribution'] == 'hdp'
+  if node['hadoop'].key?('distribution_version')
+    node['hadoop']['distribution_version']
```

Comment:
Minor suggestion: just wrap the whole thing in ``unless node['hadoop'].key?('distribution_version')`` ?  It seems a little cleaner and avoids the reassignment